dist/responsive_images.js

```diff
@@ -499,6 +510,97 @@ function getImgData( img ) {
     Object.values( imgData.intrinsicOrExtrinsicSizing ).includes( 'intrinsic' )
   );
 
+  // what was the impact of incorrect sizes values?
+
+  // helpers...
+  function geometricDistance(a, b) {
+    if (a > b) {
+      return a / b;
+    } else {
+      return b / a;
+    }
+  }
+
+  const smallestResourceThatsLargerThanDPR = function( annotatedCandidates ) {
+    if ( !annotatedCandidates || !annotatedCandidates.length || annotatedCandidates.length === 0 ) {
+      return null;
+    }
+    const larger = annotatedCandidates.filter( c => c.linearDistance >= 0 );
+    if ( larger.length === 0 ) {
+      return annotatedCandidates.sort( ( a, b ) => {
+        return a.linearDistance - b.linearDistance;
+      } )[ annotatedCandidates.length - 1 ];
+    }
+    return larger.sort( ( a, b ) => {
+      return a.linearDistance - b.linearDistance;
+    } )[ 0 ];
+  }
+
+  const resourceThatsClosestToDPR = function( annotatedCandidates ) {
+    if ( !annotatedCandidates || !annotatedCandidates.length || annotatedCandidates.length === 0 ) {
```

Comment:
Is the zero equivalence check really needed after checking `!length`?

Reply:
Nope! I'll remove.